android/app/src/main/java/com/microsoft/codepush/react/CodePushUtils.java

```diff
@@ -265,6 +271,108 @@ public static String tryGetString(ReadableMap map, String key) {
         }
     }
 
+    public static void unzipFile(File zipFile, String destination) throws IOException {
+        FileInputStream fis = null;
+        BufferedInputStream bis = null;
+        ZipInputStream zis = null;
+        try {
+            fis = new FileInputStream(zipFile);
+            bis = new BufferedInputStream(fis);
+            zis = new ZipInputStream(bis);
+            ZipEntry entry;
+
+            File destinationFolder = new File(destination);
+            if (!destinationFolder.exists()) {
+                destinationFolder.mkdirs();
+            }
+
+            byte[] buffer = new byte[WRITE_BUFFER_SIZE];
+            while ((entry = zis.getNextEntry()) != null) {
+                String fileName = entry.getName();
+                File file = new File(destinationFolder, fileName);
+                if (entry.isDirectory()) {
+                    file.mkdirs();
+                } else {
+                    File parent = file.getParentFile();
+                    if (!parent.exists()) {
+                        parent.mkdirs();
+                    }
+
+                    FileOutputStream fout = new FileOutputStream(file);
+                    try {
+                        int numBytesRead;
+                        while ((numBytesRead = zis.read(buffer)) != -1) {
+                            fout.write(buffer, 0, numBytesRead);
+                        }
+                    } finally {
+                        fout.close();
+                    }
+                }
+                long time = entry.getTime();
+                if (time > 0) {
+                    file.setLastModified(time);
+                }
+            }
+        } finally {
+            try {
+                if (zis != null) zis.close();
+                if (bis != null) bis.close();
+                if (fis != null) fis.close();
+            } catch (IOException e) {
+                throw new CodePushUnknownException("Error closing IO resources.", e);
+            }
+        }
+    }
+
+    public static void mergeEntriesInFolder(String fromPath, String destinationPath) throws IOException {
+        File fromDir = new File(fromPath);
```

Comment:
Should we go with "source" everywhere instead of "from"? I think it reads a little better.